samples/subsys/video/capture/src/main.c

```diff
@@ -78,6 +81,28 @@ int main(void)
 	       (char)(fmt.pixelformat >> 24),
 	       fmt.width, fmt.height);
 
+	if (!video_get_frmival(video, VIDEO_EP_OUT, &frmival)) {
+		printk("- Default frame rate : %f fps\n",
+		       1.0 * frmival.denominator / frmival.numerator);
+	}
+
+	printk("- Supported frame intervals for the default format:\n");
+	memset(&fie, 0, sizeof(fie));
+	fie.pixelformat = fmt.pixelformat;
+	fie.width = fmt.width;
+	fie.height = fmt.height;
+	while (video_enum_frmival(video, VIDEO_EP_OUT, &fie) == 0) {
+		if (fie.type == VIDEO_FRMIVAL_TYPE_DISCRETE) {
+			printk("   %u/%u ", fie.discrete.numerator, fie.discrete.denominator);
+		} else {
+			printk("   [min = %u/%u; max = %u/%u; step = %u/%u]\n",
+			       fie.stepwise.min.numerator, fie.stepwise.min.denominator,
+			       fie.stepwise.max.numerator, fie.stepwise.max.denominator,
+			       fie.stepwise.step.numerator, fie.stepwise.step.denominator);
+		}
+		fie.index++;
```

Comment:
FYI: there is an opportunity to reduce the drivers code size by moving these to [`drivers/video.h`](https://github.com/zephyrproject-rtos/zephyr/pull/72254/files#diff-70c4de6949b97c13ac893b4bfb16fbcc433593fbfa96ec8947bb5cb83c4d04bcR485)
In the same style as [strsep](https://www.man7.org/linux/man-pages/man3/strsep.3.html).

Reply:
I will look at this but it would be better if you could give a bit more detail

Reply:
My bad for forgetting to answer.
I was thinking of auto-incrementing `fie.index` as part of `video_enum_frmival()` or something similar, but having an explicit `fie.index++;` helps with readability.